Added ActionType service and related functionality

chaitanya-mandala opened the pull request:
- Introduced ActionType service to retrieve the type of a specified ROS action.
- Updated proxy to include logic for determining action types based on existing action servers.
- Modified rosapi_node to create the new ActionType service endpoint.
- Added ActionType message definition in rosapi_msgs.

chaitanya-mandala committed:
Added ActionType service and related functionality

- Introduced ActionType service to retrieve the type of a specified ROS action.
- Updated proxy to include logic for determining action types based on existing action servers.
- Modified rosapi_node to create the new ActionType service endpoint.
- Added ActionType message definition in rosapi_msgs.

chaitanya-mandala committed:
Fix pre-commit hook issues: end-of-file and trailing-whitespace

bjsowa commented on .vscode/settings.json:
```diff
@@ -7,7 +7,7 @@
         "editor.formatOnSave": true,
         "editor.codeActionsOnSave": {
-            "source.organizeImports": true
+            "source.organizeImports": "explicit"
```
Please revert this change

chaitanya-mandala committed:
reverted back vscode settings.json

bjsowa commented:
@chaitanya-mandala Why not use `_node.get_action_names_and_types()` instead of identifying actions by hidden topics?

chaitanya-mandala committed:
Refactor get_action_type to use get_action_names_and_types for improved action type retrieval. Removed outdated logic for action server filtering and topic name extraction.

chaitanya-mandala commented:
> @chaitanya-mandala Why not use `_node.get_action_names_and_types()` instead of identifying actions by hidden topics?


@bjsowa Thank you for the insightful suggestion! I've updated the get_action_type() function to utilize the official graph call from the ros2action library and the functionality remains consistent.

Here is test Result
```bash
chaitu@chaitu:~$ ros2 service call /rosapi/action_type rosapi_msgs/srv/ActionType '{action: "/fibonacci"}'
requester: making request: rosapi_msgs.srv.ActionType_Request(action='/fibonacci')

response:
rosapi_msgs.srv.ActionType_Response(type='action_tutorials_interfaces/action/Fibonacci')
```

chaitanya-mandala committed:
Update get_action_type function to set include_hidden parameter default to False for consistency with other functions.

bjsowa reviewed (changes requested):
Looks much simpler this way. Personally, I would rather use the method from `rclpy` directly instead of wrappers from `ros2cli` packages but since other proxy methods are also implemented this way, we can leave this and change in future code refactor.

The only change request I have is to add `ros2action` to package dependencies

chaitanya-mandala committed:
Added ros2action as a runtime dependency in package.xml

chaitanya-mandala commented:
> Looks much simpler this way. Personally, I would rather use the method from `rclpy` directly instead of wrappers from `ros2cli` packages but since other proxy methods are also implemented this way, we can leave this and change in future code refactor.
> 
> The only change request I have is to add `ros2action` to package dependencies

Thank you for your feedback! I've added ros2action dependency in the package.xml.

bjsowa committed:
Merge branch 'ros2' into ros2
